src/templates/Practice5.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from 'react';
+
+
+interface Data {
+  id: number;
+  todayDate: string;
+  taskName: string;
+  isChecked: boolean;
+}
+
+const Practice5: React.FC = () => {
+  const [todos, setTodos] = useState<Data[]>([]);
+  const [task, setTask] = useState<string>('');
+  const [allSelected, areAllSelected] = useState<boolean>(false);
+
+  const selectAll = () => {
+    const newAllSelected = !allSelected;
+    areAllSelected(newAllSelected);
+    setTodos(todos.map(todo => ({ ...todo, isChecked: newAllSelected })));
+  };
```

Comment:
Should we go to this way to toggle the state? 

```suggestion
  const selectAll = () => {
  setIsAllSelected(prevIsAllSelected => {
    const newState = !prevIsAllSelected;
    setTodos(todos => todos.map(todo => ({ ...todo, isChecked: newState })));
    return newState;
  });
};
```

Reply:
Yes we can go this way but are you suggesting it in terms of cleaner way and better readability or is this way more efficient too ?

Reply:
@awantika-b20150 For better readability! What do you think?

Reply:
Understood. I think this is better for syncing the state of our individual toggles and select all toggle. I have made the changes.